gui/src/pages/WorkspaceVariables.svelte

```diff
@@ -0,0 +1,31 @@
+<script lang="ts">
+  import Panel from '../components/Panel.svelte';
+  import Layout from '../components/Layout.svelte';
+  import Spinner from '../components/Spinner.svelte';
+  import WorkspaceNav from '../components/WorkspaceNav.svelte';
+  import EnvironmentTable from '../components/EnvironmentTable.svelte';
+  import { api } from '../lib/api';
+
+  export let params = { workspace: '' };
+
+  const workspaceId = params.workspace;
+  const workspace = api.workspace(workspaceId);
+  const workspaceRoute = `/workspaces/${encodeURIComponent(workspaceId)}`;
```

Comment:
These 3 definitions are ALL OVER the app. Almost every page has a duplication of this code. I'm not sure the best way to apply DRY here but I'm thinking maybe we should extract this to a library function that is called in a single line like

```
const { workspaceId, workspace, workspaceRoute } = someFunction(params)
```

Reply:
^ (If this is improved it probably makes sense to do it outside of this PR)

Reply:
Yeah, definitely for another PR. There are a variety of ways to DRY this up we can look at as we continue to add pages.